web/src/stream_list.ts

```diff
@@ -835,6 +838,28 @@ export function set_event_handlers({
         e.stopPropagation();
     });
 
+    $("#stream_filters").on("click", "li .subscription_block .stream-name", (e) => {
+        if (e.metaKey || e.ctrlKey) {
+            return;
+        }
+
+        const stream_id = stream_id_for_elt($(e.target).parents("li"));
+        const topics = stream_topic_history.get_recent_topic_names(stream_id);
+        const top_unmuted_topic = topics.find(
+            (topic) => !user_topics.is_topic_muted(stream_id, topic),
+        );
+
+        if (top_unmuted_topic) {
```

Comment:
We should still narrow to stream instead of doing nothing if there no unmuted topics.

Reply:
This was not correctly addressed to fixed it.